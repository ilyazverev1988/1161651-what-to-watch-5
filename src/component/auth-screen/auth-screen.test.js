import React from "react";
import renderer from "react-test-renderer";
import AuthScreen from "./auth-screen";
import {MemoryRouter} from 'react-router-dom';

let errorEmail = `Please enter a valid email address`;
let errorPassword = `Please enter a password`;
let errorAuthorization = `We canâ€™t recognize this email and password combination. Please try again.`;
const noop = () => {};

describe(`Should AuthScreen render correctly`, () => {
  it(`With no error`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <AuthScreen
              errorEmail={``}
              errorPassword={``}
              errorAuthorization={``}
              handleChangePassword={noop}
              handleChangeEmail={noop}
              handleSubmit={noop}
            />
          </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`With errorEmail`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <AuthScreen
              errorEmail={errorEmail}
              errorPassword={``}
              errorAuthorization={``}
              handleChangePassword={noop}
              handleChangeEmail={noop}
              handleSubmit={noop}
            />
          </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`With errorPassword`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <AuthScreen
              errorEmail={``}
              errorPassword={errorPassword}
              errorAuthorization={``}
              handleChangePassword={noop}
              handleChangeEmail={noop}
              handleSubmit={noop}
            />
          </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`With errorAuthorization`, () => {
    const tree = renderer
      .create(
          <MemoryRouter>
            <AuthScreen
              errorEmail={``}
              errorPassword={``}
              errorAuthorization={errorAuthorization}
              handleChangePassword={noop}
              handleChangeEmail={noop}
              handleSubmit={noop}
            />
          </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
