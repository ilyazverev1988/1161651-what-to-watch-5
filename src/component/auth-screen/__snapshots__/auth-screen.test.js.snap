// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AuthScreen render correctly With errorAuthorization 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          We can’t recognize this email and password combination. Please try again.
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            placeholder="Email address"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          disabled="We can’t recognize this email and password combination. Please try again."
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Should AuthScreen render correctly With errorEmail 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          Please enter a valid email address
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field sign-in__field--error"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            placeholder="Email address"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          disabled="Please enter a valid email address"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Should AuthScreen render correctly With errorPassword 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            placeholder="Email address"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          disabled="Please enter a password"
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Should AuthScreen render correctly With no error 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            placeholder="Email address"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          disabled=""
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
